robot/basestation/static/js/elements/navigation/localization.js

```diff
@@ -0,0 +1,129 @@
+let rover_heading = 0
+let rover_x = 0
+let rover_y = 0
+let zoom_value = 1
+let MAX_ZOOM = 2
+let MIN_ZOOM = 0.25
+let canvas_pos
+let keys = { //javascript keycodes
+    'r': 82,
+    'w': 87,
+    'a': 65,
+    's': 83,
+    'd': 68
+}
+
+let base_lat
+let base_long
+$(document).ready(() => {
+    // the elements might not be initialized yet, must test
+    base_lat = $('#antenna-latitude').text()
+    base_long = $('#antenna-longitude').text()
+})
+
+function setup() {
+    let canvas = createCanvas(450, 300)
```

Comment:
I'm surprised Cedric only commented once, because you have magic values all over this code. Magic values are when you put a constant directly in the code rather than defining a variable up top and referring to it in the code. For example, `DEFAULT_CANVAS_HEIGHT`  for a default value and `canvasHeight` for one that would change over time, instead of just sticking values right in here. You've done this in many locations in this file and possibly others.

Reply:
I'm going to comment on a bunch of the ones I find, although @MartensCedric might want to confirm the spots and point out anything I miss cause maybe I don't have the best judgement.